include/translated-post.php

```diff
@@ -259,45 +259,76 @@ public function current_user_can_read( $post_id, $context = 'view' ) {
 	 * @return WP_Post[] Array of posts.
 	 */
 	public function get_untranslated( $type, $untranslated_in, $lang, $search = '' ) {
-		$return = array();
-
-		// Don't order by title: see https://wordpress.org/support/topic/find-translated-post-when-10-is-not-enough
-		$args = array(
-			's'                => $search,
-			'suppress_filters' => 0, // To make the post_fields filter work
-			'lang'             => 0, // Avoid admin language filter
-			'numberposts'      => 20, // Limit to 20 posts
-			'post_status'      => 'any',
-			'post_type'        => $type,
-			'tax_query'        => array(
-				array(
-					'taxonomy' => $this->tax_language,
-					'field'    => 'term_taxonomy_id', // WP 3.5+
-					'terms'    => $lang->term_taxonomy_id,
-				),
-			),
-		);
+		global $wpdb;
 
 		/**
-		 * Filter the query args when auto suggesting untranslated posts in the Languages metabox
-		 * This should help plugins to fix some edge cases
+		 * Filters the limit of queried posts during a search of untranslated ones.
+		 * Formerly `pll_ajax_posts_not_translated_args` when `get_posts()` was used.
 		 *
-		 * @see https://wordpress.org/support/topic/find-translated-post-when-10-is-not-enough
+		 * @since 3.4
 		 *
-		 * @since 1.7
-		 *
-		 * @param array $args WP_Query arguments
+		 * @param int $limit Limit to searched posts, default 20.
 		 */
-		$args  = apply_filters( 'pll_ajax_posts_not_translated_args', $args );
-		$posts = get_posts( $args );
+		$limit = apply_filters( 'pll_ajax_posts_not_translated_limit', 20 );
```

Comment:
I'd like to keep the current filter. We could allow to filter only 'number_posts' for now but would be able to add new parameters if asked by users.

Reply:
I set it back in 4181a052.